src/test/java/uk/gov/digital/ho/hocs/info/api/SchemaIntegrationTest.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.digital.ho.hocs.info.api;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.http.entity.ContentType;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.SqlConfig;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.gov.digital.ho.hocs.info.api.dto.SchemaDto;
+import uk.gov.digital.ho.hocs.info.domain.model.NominatedContact;
+
+import java.io.IOException;
+import java.util.Set;
+
+import static org.springframework.test.context.jdbc.Sql.ExecutionPhase.AFTER_TEST_METHOD;
+import static org.springframework.test.context.jdbc.SqlConfig.TransactionMode.ISOLATED;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(properties = { "user.email.whitelist=homeoffice.gov.uk" },
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@Sql(scripts = "classpath:beforeTest.sql", config = @SqlConfig(transactionMode = ISOLATED))
+@Sql(scripts = "classpath:afterTest.sql", config = @SqlConfig(transactionMode = ISOLATED), executionPhase = AFTER_TEST_METHOD)
+@ActiveProfiles("test")
+public class SchemaIntegrationTest {
+
+    public static final String TEST_SCREEN_SCHEMA = "TEST_SCREEN_SCHEMA";
+    TestRestTemplate restTemplate = new TestRestTemplate();
+
+    private HttpHeaders headers;
+
+    @LocalServerPort
+    int port;
+
+    @Autowired
+    ObjectMapper mapper;
+
+    @Before
+    public void setup() throws IOException {
+        headers = new HttpHeaders();
+        headers.add(HttpHeaders.ACCEPT, ContentType.APPLICATION_JSON.toString());
+    }
+
+    @Test
+    public void shouldGetSchema() {
+        String schemaType = TEST_SCREEN_SCHEMA;
+
+        HttpEntity httpEntity = new HttpEntity(headers);
+
+        ResponseEntity<SchemaDto> result = restTemplate.exchange(
+                getBasePath() + "/schema/" + schemaType
+                , HttpMethod.GET, httpEntity, SchemaDto.class);
+
+        SchemaDto schemaDto = result.getBody();
+//        assertThat(userDto.getEmail()).isEqualTo(createUserDto.getEmail());
+//        assertThat(userDto.getFirstName()).isEqualTo(createUserDto.getFirstName());
+//        assertThat(userDto.getLastName()).isEqualTo(c
+
+
+    }
+
+    private String getBasePath() {
```

Comment:
That's interesting. I've made this changed.
---
Separate from this PR, would it be possible to configure this once in a bean and inject it into all the integration tests?

Reply:
You could... as a helper or something in the testing classpath. For the sake of the tests though I personally think it's better to try and not make the testing set up too complex. Otherwise you start needing to right tests to test the logic in your tests. So probably better to write them as you've written this one. Anyway... lgtm